src/autopas/pairwiseFunctors/Functor.h

```diff
@@ -241,7 +250,9 @@ class Functor {
    * @param offset Offset within the SoA. The data of the cell should be added
    * to the SoA with the specified offset.
    */
-  virtual void SoALoader(ParticleCell<Particle> &cell, ::autopas::SoA<SoAArraysType> &soa, size_t offset) {
+
+  template <class ParticleCell>
```

Comment:
I'm pretty sure doxygen will complain about missing `@tparam` documentation ;)
While you are at it, please remove the `@brief` keyword.